jerry-core/ecma/builtin-objects/ecma-builtin-regexp-string-iterator-prototype.c

```diff
@@ -0,0 +1,192 @@
+
+#define BUILTIN_INC_HEADER_NAME "ecma-builtin-regexp-string-iterator-prototype.inc.h"
+#define BUILTIN_UNDERSCORED_ID regexp_string_iterator_prototype
+#include "ecma-builtin-internal-routines-template.inc.h"
+
+/** \addtogroup ecma ECMA
+ * @{
+ *
+ * \addtogroup ecmabuiltins
+ * @{
+ *
+ * \addtogroup %regexpstringiteratorprototype% ECMA %ArrayIteratorPrototype% object built-in
+ * @{
+ */
+
+/**
+ * The %RegExpStringIteratorPrototype% object's 'next' routine
+ *
+ * See also:
+ *          ECMA-262 v11, 21.2.7.1.1
+ *
+ * Note:
+ *     Returned value must be freed with ecma_free_value.
+ *
+ * @return iterator result object, if success
+ *         error - otherwise
+ */
+static ecma_value_t
+ecma_builtin_regexp_string_iterator_prototype_object_next (ecma_value_t this_val) /**< this argument */
+{
+  /* 2. */
+  if (!ecma_is_value_object (this_val))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not an object."));
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (this_val);
+  ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) obj_p;
+
+  /* 3. */
+  if (ecma_get_object_type (obj_p) != ECMA_OBJECT_TYPE_PSEUDO_ARRAY
+      || ext_obj_p->u.pseudo_array.type != ECMA_PSEUDO_REGEXP_STRING_ITERATOR)
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not an iterator."));
+  }
+
+  ecma_regexp_string_iterator_t *regexp_string_iterator_obj = (ecma_regexp_string_iterator_t *) obj_p;
+
+  /* 4. */
+  if (ecma_is_value_empty (regexp_string_iterator_obj->iterated_string))
+  {
+    return ecma_create_iter_result_object (ECMA_VALUE_UNDEFINED, ECMA_VALUE_TRUE);
+  }
+
+  /* 5. */
+  ecma_value_t regexp = regexp_string_iterator_obj->iterating_regexp;
+
+  /* 6. */
+  ecma_value_t matcher_str_value = regexp_string_iterator_obj->iterated_string;
+  ecma_string_t *matcher_str_p = ecma_get_string_from_value (matcher_str_value);
+
+  /* 9. */
+  ecma_value_t match = ecma_op_regexp_exec (regexp, matcher_str_p);
+
+  if (ECMA_IS_VALUE_ERROR (match))
+  {
+    return match;
+  }
+
+  /* 10. */
+  if (ecma_is_value_null (match))
+  {
+    ecma_free_value (regexp_string_iterator_obj->iterated_string);
+    regexp_string_iterator_obj->iterated_string = ECMA_VALUE_EMPTY;
+    return ecma_create_iter_result_object (ECMA_VALUE_UNDEFINED, ECMA_VALUE_TRUE);
+  }
+
+  ecma_object_t *match_result_array_p = ecma_get_object_from_value (match);
+
+  ecma_value_t result = ECMA_VALUE_ERROR;
+
+  /* 11. */
+  if (regexp_string_iterator_obj->header.u.pseudo_array.extra_info & RE_FLAG_GLOBAL)
+  {
+    ecma_value_t matched_str_value = ecma_op_object_get_by_index (match_result_array_p, 0);
+
+    if (ECMA_IS_VALUE_ERROR (matched_str_value))
+    {
+      goto free_variables;
+    }
+
+    ecma_string_t *matched_str_p = ecma_op_to_string (matched_str_value);
+
+    ecma_free_value (matched_str_value);
+
+    if (JERRY_UNLIKELY (matched_str_p == NULL))
+    {
+      ecma_deref_ecma_string (matched_str_p);
```

Comment:
That's seems to me a NULL pointer dereference. Please add a testcase for it as well.